internal/services/runservice/common/common.go

```diff
@@ -127,6 +128,9 @@ func GenExecutorTaskSpecData(r *types.Run, rt *types.RunTask, rc *types.RunConfi
 	// run config Environment variables ovverride every other environment variable
 	mergeEnv(environment, rc.Environment)
 
+	// AGOLA_RUN_COUNTER is generated at runtime since Counter doesn't exist before creating the run
```

Comment:
This isn't really true, counter is created before creating the run.

Reply:
I mean it is not defined before call the api (it is not defined in RunCreateRequest run api types).
The counter is created in the same transaction before creating the run, but I thought it is not a good place to add variables to the run environment.

The comment could be for example `AGOLA_RUN_COUNTER is generated at runtime since Counter is not defined before the save run transaction` ?

Reply:
I think it should just be

`The AGOLA_RUN_COUNTER environment variable is not saved in the runconfig StaticEnvironment map but it populated here using the run counter.`